infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -520,6 +520,15 @@ private DetectorTask detectFailure(PollReport pollReport, Layout layout) {
         log.trace("Handle failures for the report: {}", pollReport);
 
         try {
+            //** xq check local node logunit write/read latency
+            if (IOLatencyDetector.reportSpike() == true) {
```

Comment:
I think this is good for logging purposes and metrics collection. 

As for the fault detector I think it should ultimately evaluate the end-to-end metrics and not just IO. Why not gentrify this to also capture rpc level stats?

Reply:
Will work on this for another patch.